app/models/course/material/folder.rb

```diff
@@ -3,6 +3,8 @@ class Course::Material::Folder < ActiveRecord::Base
   include Course::ModelComponentHost::Component
 
   after_initialize :set_defaults, if: :new_record?
+  before_validation :assign_valid_name
+  before_save :assign_valid_name
```

Comment:
then that's a problem with using before_create, and before_validation should be used instead?